### PR TITLE
Save app and device info to global var

### PR DESCRIPTION
**Pull request by cewert**
This saves some `roAppInfo` and `roDeviceInfo` to the global var. This makes it easier to access and prevents us from having to recreate the objects just to grab one piece of data.

NOTE: Some references to `roDeviceInfo` remain because there are certain functions being used that require access to the object. I considered just keeping the whole object in the global var but that feels like a waste of memory when we can create the object as needed for the functions that need it.

## Changes
- Save some data from `roAppInfo` to global var
- Search code base for `roAppInfo` and update to use global var
- Save some data from `roDeviceInfo` to global var
- Search code base for `roDeviceInfo` and update to use global var
- Update `m.global.device.isAudioGuideEnabled` whenever user toggles the feature
- Move `setConstants()` to `globals.brs`
- Remove unused `source\api\constants.brs` file

This PR shouldn't change any functionality besides perhaps speeding things up a bit.

### CODE REVIEW

File: source/Main.brs [hunk 1]
```diff
@@ -85,10 +117,10 @@ sub Main (args as dynamic) as void
...
     input.SetMessagePort(m.port)
 
-    m.device = CreateObject("roDeviceInfo")
-    m.device.setMessagePort(m.port)
-    m.device.EnableScreensaverExitedEvent(true)
-    m.device.EnableAppFocusEvent(false)
+    device = CreateObject("roDeviceInfo")
+    device.setMessagePort(m.port)
+    device.EnableScreensaverExitedEvent(true)
+    device.EnableAppFocusEvent(false)
```
Comment (cewert) [inline source/Main.brs:79]: `source/utils/Subtitles.brs` was the only file using the device object saved to `m.` so I updated the Subtitle file to create and use it's own object so this one doesn't get saved in memory forever.

File: components/captionTask.brs [hunk 2]
```diff
@@ -17,12 +17,12 @@ sub init()
...
-    m.textColor = m.textColorDict[m.settings.GetCaptionsOption("Text/Color")]
-    m.textOpac = m.percentageDict[m.settings.GetCaptionsOption("Text/Opacity")]
-    m.bgColor = m.bgColorDict[m.settings.GetCaptionsOption("Background/Color")]
-    m.bgOpac = m.percentageDict[m.settings.GetCaptionsOption("Background/Opacity")]
+    deviceInfo = CreateObject("roDeviceInfo")
+    m.fontSize = m.fontSizeDict[deviceInfo.GetCaptionsOption("Text/Size")]
+    m.textColor = m.textColorDict[deviceInfo.GetCaptionsOption("Text/Color")]
+    m.textOpac = m.percentageDict[deviceInfo.GetCaptionsOption("Text/Opacity")]
+    m.bgColor = m.bgColorDict[deviceInfo.GetCaptionsOption("Background/Color")]
+    m.bgOpac = m.percentageDict[deviceInfo.GetCaptionsOption("Background/Opacity")]
```
Comment (cewert) [inline components/captionTask.brs:25]: The rest of the file doesn't reference `m.settings` so I removed the object from the `m.` var to free up memory. This shouldn't affect anything but there is a chance this breaks something and should be tested.

File: source/Main.brs [hunk 3]
```diff
@@ -44,6 +44,42 @@ sub Main (args as dynamic) as void
...
+            id: deviceInfo.getChannelClientID(),
+            uuid: deviceInfo.GetRandomUUID(),
+            name: deviceInfo.getModelDisplayName(),
+            friendlyName: filteredFriendly,
+            model: deviceInfo.GetModel(),
+            modelType: deviceInfo.GetModelType(),
+            osVersion: deviceInfo.GetOSVersion(),
+            locale: deviceInfo.GetCurrentLocale(),
+            clockFormat: deviceInfo.GetClockFormat(),
+            isAudioGuideEnabled: deviceInfo.IsAudioGuideEnabled(),
```
Comment (1hitsong) [inline source/Main.brs]: I believe this is the only device info item listed that can change while Jellyfin is open. In this case, we only save the original value, and if a user toggles the audio guide, we won't recognize the new value.

File: source/api/baserequest.brs [hunk 4]
```diff
@@ -135,35 +135,20 @@ function get_url()
...
-    user = get_setting("active_user")
     if user <> invalid and user <> ""
+        auth = auth + ", DeviceId=" + Chr(34) + m.global.device.id + Chr(34)
         auth = auth + ", UserId=" + Chr(34) + user + Chr(34)
+    else
+        auth = auth + ", DeviceId=" + Chr(34) + m.global.device.uuid + Chr(34)
     end if
 
+    auth = auth + ", Version=" + Chr(34) + m.global.app.version + Chr(34)
+
```
Comment (1hitsong) [inline source/api/baserequest.brs:151]: WOW! This function is sooooo much cleaner! ❤️

File: source/Main.brs (hunk 3 above)
Comment (cewert) [reply source/Main.brs]: I didn't know that - great catch!

It looks like we could use the [EnableAudioGuideChangedEvent](https://developer.roku.com/docs/references/brightscript/interfaces/ifdeviceinfo.md#enableaudioguidechangedeventenable-as-boolean-as-dynamic) function and update the global var in the main message port loop.

**Comment by cewert** [comment]
I'm now updating the `isAudioGuideEnabled` global var whenever the user enables it on their remote. If your device supports it you can enable it by pressing the star key 4 times in a row while the app is open

global-array-audioguide-enabled

NOTE: for some reason the array stored on global turns into lower case so I updated my code to keep the `isAudioGuideEnabled` property lowercase as well (my screenshot was before my latest commit titled "keep var lowercase like the rest"

**Comment by cewert** [comment]
Oh and I also deleted the unused `Globals.brs` file. None of the three functions inside were being used in our codebase

**Comment by cewert** [comment]
I moved the main logic out of `Main.brs` and into functions that are called from `Main.brs`. This makes `Main.brs` smaller and easier to navigate, provides a place for us to manipulate the data before it's saved to global, and frees up memory by destroying local variables we no longer need i.e. `appInfo`, `deviceInfo`, `regex`, `filteredFriendly`, etc.

**Comment by 1hitsong** [comment]
Do you think we should combine `source\api\constants.brs` and `source\utils\globals.brs` since they're both writing to the global node?

**Comment by cewert** [comment]
Sounds like a good idea to me. I deleted `source\api\constants.brs` and moved `setConstants()` to `source\utils\globals.brs`

**Review by 1hitsong — APPROVED** [review]

**Pull request merged**